packages/fx-core/src/component/deploy/azureDeployDriver.ts

```diff
@@ -0,0 +1,255 @@
+    const azureResource = this.parseResourceId(resourceId);
+    const azureCredential = await this.getAzureAccountCredential(this.context.azureAccountProvider);
+
+    return await this.azureDeploy(
+      { src: src, dist: dist, ignoreFile: args.ignoreFile },
+      azureResource,
+      azureCredential
+    );
+  }
+
+  /**
+   * real azure deploy logic
+   * @param args local file needed to be deployed
+   * @param azureResource azure resource info
+   * @param azureCredential azure user login credential
+   */
+  abstract azureDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredentialsBase
+  ): Promise<void>;
+
+  /**
+   * check if resource id is legal and parse it
+   * @param resourceId deploy target
+   * @protected
+   */
+  protected parseResourceId(resourceId: string): AzureResourceInfo {
+    const result = resourceId.trim().match(this.pattern);
+    if (!result || result.length != 4) {
+      throw PrerequisiteError.somethingIllegal("resourceId", "plugins.bot.InvalidValue", [
+        "resourceId",
+        resourceId,
+      ]);
+    }
+    return {
+      subscriptionId: resourceId[1].trim(),
+      resourceGroupName: resourceId[2].trim(),
+      instanceId: resourceId[3].trim(),
+    };
+  }
+
+  /**
+   * deploy to azure app service or azure function use zip deploy method
+   * @param args local file needed to be deployed
+   * @param azureResource azure resource info
+   * @param azureCredential azure user login credential
+   * @protected
+   */
+  protected async zipDeploy(
+    args: DeployStepArgs,
+    azureResource: AzureResourceInfo,
+    azureCredential: TokenCredentialsBase
+  ): Promise<void> {
+    const zipBuffer = await this.packageToZip(args, this.context);
+    const config = await this.createAzureDeployConfig(azureResource, azureCredential);
+    const endpoint = this.getZipDeployEndpoint(azureResource.resourceGroupName);
+    const location = await AzureDeployDriver.zipDeployPackage(
+      endpoint,
+      zipBuffer,
+      config,
+      this.context.logProvider
+    );
+    await AzureDeployDriver.checkDeployStatus(location, config, this.context.logProvider);
+  }
+
+  /**
+   * call azure app service or azure function zip deploy method
+   * @param zipDeployEndpoint azure zip deploy endpoint
+   * @param zipBuffer zip file buffer
+   * @param config azure upload config, including azure account credential
+   * @param logger log provider
+   * @protected
+   */
+  protected static async zipDeployPackage(
+    zipDeployEndpoint: string,
+    zipBuffer: Buffer,
+    config: AzureUploadConfig,
+    logger?: LogProvider
+  ): Promise<string> {
+    let res: AxiosZipDeployResult;
+    try {
+      res = await AzureDeployDriver.AXIOS_INSTANCE.post(zipDeployEndpoint, zipBuffer, config);
+    } catch (e) {
+      throw DeployExternalApiCallError.zipDeployError(e);
+    }
+
+    if (res?.status !== 200) {
```

Comment:
Nit: 200 can be replaced by some constant defined previously.

Reply:
Maybe you can use the constant you defined in HttpStatusCode here?